### PR TITLE
darwin: explicitly mark broken for failed building apple packages

### PR DESCRIPTION
**Pull request by holymonson**
###### Motivation for this change
We are trying to update and clean apple packages, so we explicitly mark broken for those mean to fail building.

###### Things done
Mark `hfs` and `libauto` broken.

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Please move those two changes into two commits and name them properly. I imagine something like ``darwin.libaudio: mark broken``.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 108651` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>darwin.hfs</li>
    <li>darwin.libauto</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
@holymonson please resolve the eval error.

File: pkgs/os-specific/darwin/apple-source-releases/libauto/default.nix [hunk 1]
```diff
@@ -79,6 +79,8 @@ appleDerivation {
   '';
 
   meta = {
+    # libauro is only used by objc4/pure.nix , but objc4 is now using the impure approach, so we don't bother to fix this.
```
Comment (veprbl) [inline pkgs/os-specific/darwin/apple-source-releases/libauto/default.nix:82]: A small typo
```suggestion
    # libauto is only used by objc4/pure.nix , but objc4 is now using the impure approach, so we don't bother to fix this.
```

**Review by SuperSandro2000 — APPROVED** [review]

**Review by veprbl — APPROVED** [review]

**Pull request merged**